csrc/kernel_cache.cpp

```diff
@@ -44,9 +44,12 @@ c10::ThreadPool* getThreadPool() {
   return &pool;
 }
 
-void encodeBuffer(size_t value, std::string& buffer) {
+// Copy bytes of value to back of buffer. This is templated in order to avoid
+// implicit cast such as int64_t -> size_t that might lose information.
+template <typename T>
+void encodeBuffer(T value, std::string& buffer) {
```

Comment:
I noticed that a signed int scalar input might now be passed to this function resulting in an implicit cast to `size_t`. In order to ensure we are always appending the input byte-for-byte I templated it.